### PR TITLE
feature: first pass at hero image

### PR DESCRIPTION
**Pull request by ashleysyg**

### CODE REVIEW

File: components/base/DHero.vue [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <template>
   <section
+    v-if="
+      $route.params.main === 'services' ||
+      $route.params.main === 'our-work' ||
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:5]: Should we switch the if logic so that it looks for home in the positive case, then falls through to the plain banner?  This is probably safer maintenance-wise if we add more sections to the web site

Comment (mcmcgrath13) [reply components/base/DHero.vue:5]: also, it looks like the only difference between the if and the else is the `header-image` class, can that just be added to the `:class` binding somehow instead?

**Review by mcmcgrath13 — CHANGES_REQUESTED** [review]

**Comment by mcmcgrath13** [comment]
It looks pretty! It's a bit hard to read depending on where you are in the scroll - could you play around with a mostly transparent box around the text to see if that helps? Or other techniques?

**Review by mcmcgrath13 — APPROVED** [review]
☂️

**Review by mcmcgrath13 — COMMENTED** [review]
still generally approved, but remember to remove the ccv-sketch since it isn't used and also fix the build error

**Comment by ashleysyg** [comment]
@mcmcgrath13 I think this is ready to go, but checks are failing because of Isabel's changes to the content repo.

**Comment by mcmcgrath13** [comment]
> @mcmcgrath13 I think this is ready to go, but checks are failing because of Isabel's changes to the content repo.

I don't think that's been merged yet.  Looking at the CI logs, it's unhappy about using `rgb` with 4 arguments instead of `rgba`.

File: assets/scss/_button.scss [hunk 2]
```diff
@@ -2,15 +2,13 @@
 
 .d-button {
   @extend .button;
```
Comment (mcmcgrath13) [inline assets/scss/_button.scss:4]: not on this file, but I don't know how to comment on binary files.  I don't think ccv-sketch is being used anywhere -delete? (and generally any other files not in use)

File: components/base/DHero.vue [hunk 3]
```diff
@@ -2,17 +2,22 @@
   <section
     role="banner"
     class="d-hero is-medium"
-    :class="['is-' + variant, { 'is-full-height': fullHeight }]"
+    :class="[
+      'is-' + variant,
+      { 'is-full-height': fullHeight, 'header-image': !$route.params.main, 'header-image-alt': $route.params.main },
+    ]"
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:8]:
```suggestion
    :class="[
      'is-' + variant,
      $route.params.main ? 'header-image-alt' : 'header-image',
      { 'is-full-height': fullHeight },
    ]"
```

File: components/base/DHero.vue [hunk 4]
```diff
@@ -39,3 +44,31 @@ export default {
...
+  background-size: cover;
+  background-color: #999;
+
+  .hero-text {
+    background-color: rgb(0, 0, 0, 0.35);
+  }
+}
+
+.header-image-alt {
+  background-image: url('@/assets/stone_hero.jpeg');
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:63]: I thought we were going with the swirly blue one - why the change?

File: components/base/DHero.vue [hunk 5]
```diff
@@ -39,3 +44,31 @@ export default {
...
+.header-image {
+  background-image: url('@/assets/ccv-original.jpg');
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-attachment: fixed;
+  background-size: cover;
+  background-color: #999;
+
+  .hero-text {
+    background-color: rgb(0, 0, 0, 0.35);
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:58]:
```suggestion
    background-color: rgba(0, 0, 0, 0.35);
```

File: components/base/DHero.vue [hunk 6]
```diff
@@ -39,3 +44,31 @@ export default {
...
+.header-image-alt {
+  background-image: url('@/assets/stone_hero.jpeg');
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-attachment: fixed;
+  background-size: cover;
+  background-color: #999;
+
+    .hero-text {
+    background-color: rgb(0, 0, 0, 0.35);
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:71]:
```suggestion
    background-color: rgba(0, 0, 0, 0.35);
```

File: components/base/DHero.vue [hunk 7]
```diff
@@ -39,3 +44,31 @@ export default {
   },
 };
 </script>
+
+<style lang="scss" scoped>
+.header-image {
```
Comment (mcmcgrath13) [inline components/base/DHero.vue:49]: it seems most of this and `.header-image-alt` are the same.  Maybe make three classes `header-image`, which is the common things, then `image-home` and `image-alt` or something for the actual images?

File: components/blocks/FilesToCards.vue [hunk 8]
```diff
@@ -10,8 +10,7 @@
         card-container
         is-flex
         mt-6
-        is-justify-content-space-evenly 
-        is-flex-wrap-wrap
+        is-justify-content-space-evenly is-flex-wrap-wrap
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:13]: did the linter do this? it looks kind of funny to have most classes one per line, then this one two per line

File: components/blocks/FilesToCards.vue [hunk 9]
```diff
@@ -62,89 +61,102 @@
         </template>
       </DCard>
     </div>
-    <div v-else class='container'>
-      <div class="dropdown is-flex is-flex-wrap-wrap is-justify-content-space-evenly">
+    <div v-else class="container">
+      <div
+        class="
+          dropdown
+          is-flex is-flex-wrap-wrap is-justify-content-space-evenly
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:68]: same as above

File: components/blocks/FilesToCards.vue [hunk 10]
```diff
@@ -62,89 +61,102 @@
...
               </abbr></span
             >
-            <h2 class="card-title">{{item.title}}</h2>
+            <h2 class="title has-text-black pt-3">{{ item.title }}</h2>
             <div v-if="item.date">Updated: {{ item.date }}</div>
-            <span v-if="item.authors" class="small has-text-link"
-              >Authors: {{ item.authors.map((a) => a.name).join(', ') }}</span
+            <span v-if="item.authors" class="subtitle has-text-black"
+              ><i class="mdi mdi-account-multiple p-1 m-1"></i
+              >{{ item.authors.map((a) => a.name).join(', ') }}</span
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:135]: is there also a github link already? or is that being added? if so, should render conditionally as a link if available

File: components/blocks/FilesToCards.vue [hunk 11]
```diff
@@ -62,89 +61,102 @@
...
+            <section v-if="item.links" class="link-group">
+              <div><i class="mdi mdi-link p-1 title"></i></div>
+              <a
+                v-for="(link, type) in item.links"
+                :key="type"
+                class="
+                  m-1
+                  link-item
+                  d-button
+                  has-background-link has-text-white has-text-weight-semibold
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:151]: same question on one per line and linting

**Review by mcmcgrath13 — CHANGES_REQUESTED** [review]

File: components/blocks/FilesToCards.vue (hunk 10 above)
Comment (ashleysyg) [reply components/blocks/FilesToCards.vue:135]: @mcmcgrath13 a github link to the authors?

Comment (mcmcgrath13) [reply components/blocks/FilesToCards.vue:135]: yup! I think it gives username, but that's easily turned into a link

**Comment by ashleysyg** [comment]
Screen Shot 2021-10-10 at 9 17 04 AM
Screen Shot 2021-10-10 at 9 44 12 AM
@mcmcgrath13 @mirestrepo @paulstey do we like this background image? which version?

**Comment by mirestrepo** [comment]
I like them both with a slight preference towards the last one (black text)

**Comment by mcmcgrath13** [comment]
I also prefer the black text one - what do you think about adding a slight white background to it:

image

background is `hsla(0, 0, 100%, 0.2)`, color is `hsla(0, 0, 20%, 0.9)` in this image

File: assets/scss/_button.scss [hunk 12]
```diff
@@ -2,15 +2,13 @@
...
-  border-radius: 0 !important;
   font-family: $font-family-bold;
   white-space: normal;
```
Comment (mcmcgrath13) [inline assets/scss/_button.scss:6]: delete unused background file in assets (can't comment on those files...)

File: components/blocks/FilesToCards.vue [hunk 13]
```diff
@@ -119,41 +129,51 @@
...
             <div v-if="item.date">Updated: {{ item.date }}</div>
-            <span v-if="item.authors" class="small has-text-link"
-              >Authors: {{ item.authors.map((a) => a.name).join(', ') }}</span
-            >
+            <span v-if="item.authors" class="subtitle has-text-black"
+              ><i class="mdi mdi-account-multiple p-1 m-1"></i>
+              <span
+                v-for="(author, index) in item.authors"
+                :key="author.github_user"
+                ><a :href="'https://github.com/' + author.github_user">{{
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:148]: I don't think the github_user is guaranteed/required, so should probably key on name and then use v-if/else for the a vs span

File: components/blocks/FilesToCards.vue [hunk 14]
```diff
@@ -119,41 +129,51 @@
...
+                  author.name
+                }}</a
+                ><span v-if="index + 1 < item.authors.length">, </span></span
+              >
```
Comment (mcmcgrath13) [inline components/blocks/FilesToCards.vue:152]:
```suggestion
              <span
                v-for="(author, index) in item.authors"
                :key="author.name"
                ><a v-if="author.github_user :href="'https://github.com/' + author.github_user">{{
                  author.name
                }}</a
                >{{ !author.github_user && author.name }}<span v-if="index + 1 < item.authors.length">, </span></span
              >
```

Yup! You can combine, just not on the same element, so you have to do something like the above.

**Review by mcmcgrath13 — APPROVED** [review]
🥇
